plugins/frontend/backstage-plugin-github-pull-requests/src/api/GithubPullRequestsClient.ts

```diff
@@ -16,48 +16,43 @@
 
 import { GithubPullRequestsApi } from './GithubPullRequestsApi';
 import { Octokit } from '@octokit/rest';
-import { PullsListResponseData } from '@octokit/types';
-import { PullRequestState } from '../types';
+import { PullRequestState, SearchPullRequestsResponseData } from '../types';
 
 export class GithubPullRequestsClient implements GithubPullRequestsApi {
   async listPullRequests({
+    search = '',
     token,
     owner,
     repo,
+    defaultFilter = '',
     pageSize = 5,
     page,
     state = 'all',
     baseUrl,
   }: {
+    search: string;
     token: string;
     owner: string;
     repo: string;
+    defaultFilter: string;
     pageSize?: number;
     page?: number;
     state?: PullRequestState;
     baseUrl: string | undefined;
   }): Promise<{
-    maxTotalItems?: number;
-    pullRequestsData: PullsListResponseData;
+    pullRequestsData: SearchPullRequestsResponseData;
   }> {
     const pullRequestResponse = await new Octokit({
       auth: token,
       ...(baseUrl && { baseUrl }),
-    }).pulls.list({
-      repo,
+    }).search.issuesAndPullRequests({
+      q: `${search} in:title type:pr state:${state} ${defaultFilter} repo:${owner}/${repo}`,
       state,
       per_page: pageSize,
       page,
-      owner,
     });
-    const paginationLinks = pullRequestResponse.headers.link;
-    const lastPage = paginationLinks?.match(/\d+(?!.*page=\d*)/g) || ['1'];
-    const maxTotalItems = paginationLinks?.endsWith('rel="last"')
-      ? parseInt(lastPage[0], 10) * pageSize
-      : undefined;
     return {
-      maxTotalItems,
```

Comment:
not used anymore?

Reply:
yes it was a bad value calculate by the api call function it was page multiply by item per page and in the new endpoint to have pull request we have the real total value